used es6 conventions for variables and anonymous functions declaration

ruggertech opened the pull request:
##### Checklist
- [x] `make -j8 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [x] commit message follows commit guidelines

##### Description of change
changed es5 "var" to "const" and anonymous functions to arrow function syntax

mscdex commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/4875/

Trott commented:
If we're going to refactor the test for mostly stylistic reasons, I'd ask that we also change the three instances of `assert.equal()` to `assert.strictEqual()`.

cjihrig commented:
I agree with @Trott. I think the `process.nextTick()` that surrounds those assertions should also have a `common.mustCall()`.

ruggertech commented:
done

santigimeno commented on test/parallel/test-fs-read-stream-err.js:
```diff
@@ -1,43 +1,43 @@
-    assert.equal(err_, err);
+  process.nextTick(() => {
+    assert.strictEqual(stream.fd, null);
+    assert.strictEqual(err_, err);
```
Use `common.mustCall()` to make sure those assertions a run, as @cjihrig suggested

sam-github commented:
const arguably does something useful, es6 functions don't (unless you are trying to access the enclosing `this`). I don't really mind all the code churn, but if you are going to change some functions to es6, please do them all, so the file is internally consistent.

sam-github commented on test/parallel/test-fs-read-stream-err.js:
```diff
@@ -1,43 +1,43 @@
-  process.nextTick(function() {
-    assert.equal(stream.fd, null);
-    assert.equal(err_, err);
+  process.nextTick(() => {
```
I see your thinking, but think its too subtle, function names don't matter, particularly not in tests. Better change them all.

ruggertech commented:
I changed all the functions in the code except for the function declaration on line 11 which was originally set as a **named** function, for the sake of easier debugging, by the original developer.  I changed all of the anonymous functions to their respective es6 arrow function form, but did not want to affect code stability or ease of debugging by changing line 11.

Should line 11 also be changed?

sam-github commented:
I think none should be changed, but if we are going to go all es6y, lets go all the way, not half way.

I'm a bit worried that the wave of "name functions for readability of stack traces" is going to crash against the wave of "use the new es6 function syntax", and leave confusion, but for now, all es6 in this test is fine.

ruggertech commented:
I see your point.  Removed the named function to maintain consistency across this file.

targos commented on test/parallel/test-fs-read-stream-err.js:
```diff
@@ -1,43 +1,43 @@
 
-stream.on('error', common.mustCall(function errorHandler(err_) {
+stream.on('error', common.mustCall((err_) => {
   console.error('error event');
```
I think this line can be removed

gibfahn replied:
@ruggertech Do you agree with @targos 's suggestion here?

ruggertech replied:
which line number?

ruggertech commented:
do we need anything else for this to be merged ?

ruggertech commented on test/parallel/test-fs-read-stream-err.js:
```diff
@@ -1,43 +1,43 @@
 
-stream.on('data', function(buf) {
+stream.on('data', (buf) => {
   stream.on('data', common.fail);  // no more 'data' events should follow
```
Something like that:

> stream.on('data', () => {
>   stream.on('data', () => common.fail('no more "data" events should follow'));  // no more 'data' events should follow
> });

??

ruggertech replied:
done

sam-github commented:
commit message doesn't follow guidelines, it's too long and doesn't start with `test: `, see https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#step-3-commit

bjouhier commented on test/parallel/test-fs-read-stream-err.js:
```diff
@@ -1,43 +1,43 @@
-stream.on('data', function(buf) {
-  stream.on('data', common.fail);  // no more 'data' events should follow
+stream.on('data', () => {
+  stream.on('data', () => common.fail);  // no more 'data' events should follow
```
should be either `() => common.fail()` or just `common.fail`.

ruggertech commented:
can this be merged?

ruggertech commented:
would appreciate if someone could merge this.

cjihrig commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/4993/

ruggertech commented:
Can someone merge please :-)

targos commented on test/parallel/test-fs-read-stream-err.js:
```diff
@@ -1,43 +1,42 @@
-    var cb = arguments[arguments.length - 1];
-    process.nextTick(function() {
+  fs.read = common.mustCall(() => {
+    const cb = arguments[arguments.length - 1];
```
`arguments` now refers to the outer function's arguments. It doesn't change the outcome of the test but I think it's better not to change this one to an arrow function.

ruggertech commented:
I have no idea how this pull request got closed...I did not close it explicitly.
Can this be undone?

targos commented:
Thanks. LGTM. Final CI: https://ci.nodejs.org/job/node-test-pull-request/5155/

jasnell commented:
Landed in ef74e03.
